### PR TITLE
Improved EPvalidator to be able to check referenced nodes

### PR DESCRIPTION
**Pull request by yzhou601**
Allowed EPvalidator to check referenced node, eg. only check referenced distribution system element for heating/cooling system.

### CODE REVIEW

File: measure.rb [hunk 1]
```diff
@@ -2835,25 +2829,6 @@ def self.add_ceiling_fans(runner, model, building, weather)
                             @cfa, @living_space)
   end
 
-  def self.check_distribution_system(building, system_values)
```
Comment (yzhou601) [inline measure.rb]: New EPvalidator will check the id reference as well as distribution system type, this test can be removed, right?

Comment (shorowit) [reply measure.rb]: Seems like it. 🚀

**Comment by yzhou601** [comment]
Didn't see available syntax to directly specify child XPath with parent node in its predicates( `current()` looks not supported yet). 
Is the current approach of using XPath variables acceptable?

File: resources/EPvalidator.rb [hunk 2]
```diff
@@ -694,12 +694,10 @@ def self.run_validator(hpxml_doc)
...
-            # check referenced node by using parent's attribute(idref) value. Parameter name follows $ChildElementName_AttributeName
-            if child.include? "$"
-              var_name = child.split("$")[1].split(/\W+/)[0]
-              element_name = var_name.split("_")[0]
-              attribute_name = var_name.split("_")[1]
-              var[var_name] = REXML::XPath.first(parent_element, element_name).attributes[attribute_name] unless REXML::XPath.first(parent_element, element_name).nil?
+            # check referenced HVACDistribution
+            if child.include? "$dist_idref" and not REXML::XPath.first(parent_element, "DistributionSystem").nil?
+              # give id value to variable $dist_idref
+              var["dist_idref"] = REXML::XPath.first(parent_element, "DistributionSystem").attributes["idref"]
```
Comment (yzhou601) [inline resources/EPvalidator.rb]: One last simplification before reverting.

**Comment by shorowit** [comment]
I think it's awesome that you found a xpath-based solution, but have to admit that I find it to be a bit opaque/complex. I'm going to close this in lieu of your other, simpler approach.

**Comment by yzhou601** [comment]
> I think it's awesome that you found a xpath-based solution, but have to admit that I find it to be a bit opaque/complex. I'm going to close this in lieu of your other, simpler approach.

Sounds good. Understood the preference of not taking Xpath adventures if we can.

**Pull request closed without merging**